### PR TITLE
Disable entity management

### PR DESCRIPTION
**Pull request by amimimor**
# Description

* conditional creation of sns topic, sns-sqs subscription and sqs queues (here forth: assets)
* mutex locks protecting concurrent r/w access to cached assets references
* major refactoring (aws policy to a file, additional consts, similar assets getOrCreate handling)
* uncaught bugfix: when using deadletters-queue, if for some reason a message was received more times than it should have been, given it should have already been moved to the deadletters-queue, we used to process it as is without any notice - effectively breaking the expected behavior the user defined

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1313

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:
* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#2035

### CODE REVIEW

File: pubsub/aws/snssqs/snssqs.go [hunk 1]
```diff
@@ -431,25 +459,61 @@ func (s *snsSqs) createQueue(queueName string) (*sqsQueueInfo, error) {
...
+		queueInfo, err = s.getQueueArn(sanitizedName)
+		if err != nil {
+			s.logger.Errorf("error fetching info for queue %s: %w", queueName, err)
+
+			return nil, err
+		}
 	}
 
+	s.queuesLock.Lock()
 	s.queues[queueName] = queueInfo
```
Comment (daixiang0) [inline pubsub/aws/snssqs/snssqs.go:515]: How about use `sync.Map`?

Comment (amimimor) [reply pubsub/aws/snssqs/snssqs.go:515]: Good option. I just followed the implementation of https://github.com/dapr/components-contrib/blob/cac03dbe874f6aee9148dc3002b734b8ea5d9c4f/pubsub/azure/servicebus/servicebus.go#L88 but I prefer doing things better

**Comment by amimimor** [comment]
> Please fix conflicts.

sorry, can you elaborate (maybe you meant merge conflicts, if so they are resolved) ?

**Comment by daixiang0** [comment]
@amimimor yes, you are done. Please sign your commits to fix DCO issue.

**Comment by mukundansundar** [comment]
@amimimor why is there 73 commits in this PR? 
Also please add `--signoff` argument on commit, for DCO check to pass.

**Comment by amimimor** [comment]
> @amimimor why is there 73 commits in this PR? 
> 
> Also please add `--signoff` argument on commit, for DCO check to pass.

I see 31 commits but why does it matter if they are squashed upon merge to upstream? Regarding the DCO, I'm in the midst of a battle with git as the signed-off-by was never a criteria in past PR so I need to amend past commits (rebase nightmare or something really clever). I think that this is missing from the contributions guide and now I'm in a rough place because of that

**Comment by amimimor** [comment]
@mukundansundar @daixiang0 I'm closing this PR because I trashed the git refs and I'd open a new one once I managed the  DCO issues. Thanks for your time and sorry for the bother!

**Pull request closed without merging**